src/mscorlib/shared/System/Span.NonGeneric.cs

```diff
@@ -22,6 +23,90 @@ namespace System
     /// </summary>
     public static class Span
     {
+        /// <summary>
+        /// Determines whether the beginning of the span matches the specified value when compared using the specified comparison option.
+        /// </summary>
+        public static bool StartsWith(this ReadOnlySpan<char> span, ReadOnlySpan<char> value, StringComparison comparison)
+        {
+            if ((uint)(comparison - StringComparison.CurrentCulture) > (StringComparison.OrdinalIgnoreCase - StringComparison.CurrentCulture))
```

Comment:
Any downside to moving this inside the length check below? e.g.

```c#
if (span.Length < value.Length)
{
    if ((uint)(comparison - StringComparison.CurrentCulture) > (StringComparison.OrdinalIgnoreCase - StringComparison.CurrentCulture))
    {
        throw new ArgumentException(SR.NotSupported_StringComparison, nameof(comparison));
    }

    return false;
}
```

Reply:
I wanted to maintain the code shape (borrowed from string). But I see your point. Since the switch statement already has a default which throws, we don't need to check this ahead of time.

Reply:
With the latest change, this feedback is no longer applicable.

Reply:
There's now an `if (value.Length == 0)` check. Move it inside that to avoid the unnecessary check in most cases?